src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -90,6 +94,26 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
         Picture picture = personToEdit.getPicture().orElse(null);
         List<SpecialDate> dates = personToEdit.getDates();
         List<Meeting> meetings = personToEdit.getMeetings();
+
+        Optional<Meeting> earliestMeeting = meetings.stream().min(Comparator.comparing(Meeting::getDate));
+        Optional<SpecialDate> earliestDate = dates.stream().min(Comparator.comparing(SpecialDate::getDate));
+
+        if (earliestMeeting.isPresent() && earliestMeeting.get().getDate().isAfter(updatedBirthday.getDate())) {
+            throw new IllegalValueException(String.format(
+                    Messages.MESSAGE_BIRTHDAY_CONSTRAINT,
+                    DateUtil.toUi(updatedBirthday.getDate()),
```

Comment:
Use `DateUtil.toErrorMessage(updatedBirthday.getDate())`, the format for this date will match user input

Reply:
A bit weird to say "Invalid date. Person is not yet born on %1$s!" when you are changing the birthday itself